forte/data/extractor/feature.py

```diff
@@ -0,0 +1,108 @@
+#  Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Any, Tuple, Union
+
+
+class Feature:
+    def __init__(self,
+                 data: List[Any],
```

Comment:
How about we directly use `ndarray` in numpy for both data and pad_id? Is it because we have type restrictions for a `ndarray`? or because `ndarray` needs to be of square shape.

Reply:
Yes, the concern is that ndarray does not support different length along a certain dimension. Here the data is of type `List[Any]` so that for each dimension, it can have different length of sub list as inner dimension features. Is there any better type for representing this?

Reply:
if it works we can stick with this.